storage/remote/queue_manager.go

```diff
@@ -234,7 +236,7 @@ type QueueManager struct {
 }
 
 // NewQueueManager builds a new QueueManager.
-func NewQueueManager(reg prometheus.Registerer, logger log.Logger, walDir string, samplesIn *ewmaRate, cfg config.QueueConfig, externalLabels labels.Labels, relabelConfigs []*relabel.Config, client StorageClient, flushDeadline time.Duration) *QueueManager {
+func NewQueueManager(reg prometheus.Registerer, logger log.Logger, walDir string, samplesIn *ewmaRate, cfg config.QueueConfig, externalLabels labels.Labels, relabelConfigs []*relabel.Config, client StorageClient, flushDeadline time.Duration, shardUpdateDuration time.Duration) *QueueManager {
```

Comment:
It doesn't look like you are actually setting `shardUpdateDuration` on the `QueueManager` anywhere.

Reply:
I'm not in this PR, but I wanted to and was trying to with the test I was talking about in the original PR message. I wanted a test that went through the whole motion of samples being written to the WAL, read from the WAL and queued up, and actually sent. Without lowering some values, such as shard update duration from 10s (to 1s for example) the test would need to run for minutes to generate useful data.

Reply:
I would say wait until there is a working test using the change before committing it. Otherwise it is a lot of changes that may or may not actually get used in the near future.